### PR TITLE
Release 0.7.0

### PR DESCRIPTION
**Pull request by edburyenegren-okta**
### Added
- `vuepress-theme-odyssey` is a new private package containing our brand new docs theme
- adds `prism-theme-odyssey` package for styling code blocks
- ESLint has been added to our test suite
- [Docs] Field Labels: replaces Form documentation and guidelines
- Turquoise 400 has been added to the color map
- Links: added support for Links with Icons
- dfn: styling and minor guidance for the HTML element
- figure/figcaption: styling and minor guidance for the HTML element
- output: styling and minor guidance for the HTML element
- q: styling and minor guidance for the HTML element
- samp: styling and minor guidance for the HTML element

### Changed
- [Docs] Content and examples for all components have been rewritten
- Incosolata is now the default monospace font for Odyssey
- External link indicator updated
- Modal: z-index has been updated
- Tabs: now uses 16px base font instead of 14px
- Tabs: Border now lives on the tablist, rather than the tabpanel
- Tabs: `tabpanel` padding updated to match the design
- Tabs: removed unnecessary `z-index` declaration
- Tag: `tag--list` now inline-block by default
- Tag: Background color updated to Gray 100
- Toast/Modal: Close UI indicator updated to use variant from odyssey-icons

### Removed
- `vuepress-theme-nimatron` has been removed in favor of `vuepress-theme-odyssey`
- [Docs] Form docs have been removed in favor of Field Labels
- Tabs: unused CSS custom properties that were used for moving tab indicator implementation have been removed

### Fixed
- Tables and Table wrappers now correctly remove margin if they are the last or only child

### CODE REVIEW

**Review by jmelberg-okta — COMMENTED** [review]
Looks like we might be missing doc on a few breaking changes + updates.

Also - please make sure to run the necessary commands to [publish a new version](https://github.com/okta/odyssey#publishing-a-new-version).

File: packages/odyssey/src/scss/abstracts/_tokens.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/abstracts/_tokens.scss:17]: We do: https://github.com/okta/odyssey/pull/832/files#diff-bd531458988753b5aeabc4d2060e177abebff0839c8c82b66572502a7915cac8R25

File: packages/odyssey/src/scss/base/_typography-text.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/base/_typography-text.scss:131]: This isn't new, it's moved.

File: packages/odyssey/src/scss/components/_checkbox.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/components/_checkbox.scss:42]: This was added for demo purposes. The format matches that of the other Button/Input state classes until Storybook is added.

File: packages/odyssey/src/scss/abstracts/_tokens.scss
Comment (jmelberg-okta) [reply packages/odyssey/src/scss/abstracts/_tokens.scss:17]: Ah - missed it! Thanks.

File: packages/odyssey/src/scss/components/_tab.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/components/_tab.scss:18]: These were deprecated but not removed in a previous release: https://github.com/okta/odyssey/pull/832/files#diff-bd531458988753b5aeabc4d2060e177abebff0839c8c82b66572502a7915cac8R39

File: packages/odyssey/src/scss/base/_typography-text.scss
Comment (jmelberg-okta) [reply packages/odyssey/src/scss/base/_typography-text.scss:131]: My bad - I see them now. 👍

File: packages/odyssey/src/scss/base/_iconography.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/base/_iconography.scss:9]: Noted in the original PR, we aren't doccing anything RTL until we have better support. This was included to ensure Grid would be a workable solution moving forward.

**Comment by edburyenegren-okta** [comment]
Version bumped @jmelberg-okta! Thanks for the callout, I was waiting on a few final fixes before "sealing" the version in, as it were.

File: packages/odyssey/src/scss/components/_modal.scss
Comment (edburyenegren-okta) [reply packages/odyssey/src/scss/components/_modal.scss:18]: Good call. It shouldn't affect anything system-official, but any `z-index` change is probably breaking until we have the guidelines locked in.

File: packages/docs/.vuepress/config.js
Comment (jmelberg-okta) [inline packages/docs/.vuepress/config.js:11]: Does this need to change to <https://odyssey.okta.design/>?

File: packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js
Comment (santhoshbalakrishnan) [inline packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js:5]: Why?

File: packages/vuepress-theme-odyssey/styles/index.scss
Comment (santhoshbalakrishnan) [inline packages/vuepress-theme-odyssey/styles/index.scss:107]: Is this going to be set back in the future?

File: packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss
Comment (santhoshbalakrishnan) [inline packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss:107]: Is ms intended for positioning like this?

File: packages/odyssey/src/scss/components/_modal.scss
Comment (jmelberg-okta) [reply packages/odyssey/src/scss/components/_modal.scss:18]: Does it make sense to hold off on this change until we finalize our guidelines on this? Otherwise isn't it likely that we'll introduce another breaking `z-index` change in a future release?

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [inline packages/odyssey/CHANGELOG.md:37]: Add `[docs]` here?

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [inline packages/odyssey/CHANGELOG.md:12]: These are [docs] only?

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [inline packages/odyssey/CHANGELOG.md:10]: In general, can we separate out `[docs]` as it does not affect any downstream consumers?

Comment (edburyenegren-okta) [reply packages/odyssey/CHANGELOG.md:10]: We've had requests on both sides of this. In the future, I'm fine separating things out, but then we'll need to separate their versioning as well. As it stands, Odyssey's Workspace has one changelog and one version.

File: packages/odyssey/CHANGELOG.md
Comment (edburyenegren-okta) [reply packages/odyssey/CHANGELOG.md:12]: @arnoldsandoval-okta and I are currently discussing collapsing all docs packages into a single `odyssey-website` package. For now, these are distinct on purpose as we've had requests for separate Docs content, VuePress Theme, and Prism theme packages. 

Prism (or its replacement) will likely stay distinct afterward as it can be applied to non-docs code as well.

File: packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss
Comment (edburyenegren-okta) [reply packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss:107]: Nope. @arnoldsandoval-okta I'm okay leaving this for release since it's docs-only. Can we log a ticket to use `calc()` here instead?

File: packages/vuepress-theme-odyssey/styles/index.scss
Comment (edburyenegren-okta) [reply packages/vuepress-theme-odyssey/styles/index.scss:107]: @arnoldsandoval-okta, i think this was your call - context?

File: packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js
Comment (edburyenegren-okta) [reply packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js:5]: cc @arnoldsandoval-okta

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [reply packages/odyssey/CHANGELOG.md:10]: I am fine with one versioning/changelog, but can we separate out docs into a separate section, so that actual consumers don't have to worry about that part? makes for easier reading and lesser confusion. As it stand, for anyone unaware, they won't know what `adds "prism-theme-odyssey" package for styling code blocks` or other doc items actually mean and what impact it has on downstream consumers.

Comment (edburyenegren-okta) [reply packages/odyssey/CHANGELOG.md:10]: Is this a blocking change? I'm up for reformatting our CHANGELOG, but we'll also need to back-update the release notes in GH to have parity. I'd rather not rush that as we're trying to get this out.

Comment (santhoshbalakrishnan) [reply packages/odyssey/CHANGELOG.md:10]: As a compromise, can we move all the `[docs]` items as the last ones within each section - added, removed, changed? Rather than having them in between component changes.

Comment (edburyenegren-okta) [reply packages/odyssey/CHANGELOG.md:10]: Updated!

Comment (edburyenegren-okta) [reply packages/odyssey/CHANGELOG.md:10]: I've also opened a ticket, so we don't forget to address this as we rethink our packages: https://github.com/okta/odyssey/issues/835

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [reply packages/odyssey/CHANGELOG.md:12]: Can we move these `-theme-` parts to the bottom as well?

File: packages/vuepress-theme-odyssey/styles/index.scss
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/styles/index.scss:107]: Yes! Currently these dont work because we have content in hidden divs. Its weird, they will come back though!

File: packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/styles/templates/_DocsTemplateComponent.scss:107]: It was either this or a hard value, ms felt like an OK (although not recommended) way to approach.

File: packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js
Comment (arnoldsandoval-okta) [reply packages/docs/.vuepress/plugins/plugin-micromodal/clientRootMixin.js:5]: It's hacky for sure @santhoshbalakrishnan and on my list to fix. Due to the way Vue rendered the page, it wasn't instantiating the external dependency on `mount,` nor did it happen on `update`. The set timeout merely ensures that the dom nodes required by the dependencies present at the time of instantiation.

**Comment by johnintrater-okta** [comment]
Looks good guys! @jmelberg-okta  anything else on your end? Would really like to get this out the door.

File: packages/vuepress-theme-odyssey/components/DocsNav.vue
Comment (jmelberg-okta) [inline packages/vuepress-theme-odyssey/components/DocsNav.vue:24]: Because we have multiple instances of `DocsNav` on the page, this `id` will **not** [be unique](https://dequeuniversity.com/rules/axe/4.0/duplicate-id-active?application=AxeChrome).

As an example, take a look at the `id` values generated on **Base** and **Updates** once this renders.

**Review by jmelberg-okta — CHANGES_REQUESTED** [review]
Few accessibility concerns - once those are addressed I think we're good to go!

| Violation | Image |
| ------- | ----- |
| Links with images are not setting `aria-hidden` correctly | Screen Shot 2020-11-20 at 8 48 20 AM |
| Tabs explain text will not pass color contrast | Screen Shot 2020-11-20 at 8 52 08 AM |
| When code is displayed, the element tags will not pass color contrast | Screen Shot 2020-11-20 at 8 53 34 AM |

File: packages/odyssey/CHANGELOG.md
Comment (jmelberg-okta) [reply packages/odyssey/CHANGELOG.md:10]: +1. It's confusing that we're coupling all CHANGELOG entries within `packages/odyssey`. Instead, each package should have their own CHANGELOG.

**Comment by edburyenegren-okta** [comment]
@jmelberg-okta a11y notes:

> Links with images are not setting `aria-hidden` correctly

While some a11y inspectors may show the SVG as invalid here, the `span` has `aria-hidden`, so its children should behave appropriately in browsers and screen readers. Testing on my end shows no unexpected behavior.

I've added `role="presentation"` to the assets themselves, so `aria-hidden` is now a fallback. This should pass automated and manual testing now.

> Tabs explain text will not pass color contrast

The content isn't intended to be readable by users of any ability, so contrast requirements don't apply here. This style was chosen intentionally as any "valid" content might be perceived as non-FPO to designers. Other examples may use this in the future with the gradient only applying to non-necessary content.

> When code is displayed, the element tags will not pass color contrast.

Correct, this has been an on and off issue since Nim. 😞  We have a ticket to address this in the December fast-follow, but Prism themes were outside the scope of compliance as we awaited an updated tertiary palette.

File: packages/vuepress-theme-odyssey/components/DocsNav.vue
Comment (arnoldsandoval-okta) [reply packages/vuepress-theme-odyssey/components/DocsNav.vue]: Great catch. As a quick fix, I have fixed this by concatenating the `variant` to the id. This works fine for our docs site because there are only two variants, but isn't optimal. Given that we're only using this on the docs site, it should be good to go!

**Comment by jmelberg-okta** [comment]
> Correct, this has been an on and off issue since Nim. 😞 We have a ticket to address this in the December fast-follow, but Prism themes were outside the scope of compliance as we awaited an updated tertiary palette.

What is the level of effort to fix this? Could we make sure this becomes a release blocker for `0.8.0`?

**Review by jmelberg-okta — APPROVED** [review]
Overall LGTM given the already approved PRs.

File: packages/vuepress-theme-odyssey/styles/index.scss
Comment (santhoshbalakrishnan) [reply packages/vuepress-theme-odyssey/styles/index.scss:107]: Should we just remove this from this release then?

File: packages/odyssey/CHANGELOG.md
Comment (santhoshbalakrishnan) [reply packages/odyssey/CHANGELOG.md:38]: Ping ^

**Comment by edburyenegren-okta** [comment]
>What is the level of effort to fix this? Could we make sure this becomes a release blocker for 0.8.0?

This feels legit to me! I don't think it's a large lift, but we'll have a little politicking to do with Brand. I've marked the existing issue as High Prio, so it'll be first on the list during planning.

Do you think the Prism package might see adoption outside of Docs? I'm 💯 on fixing this just for ODS, but we'll be able to convince others of priority if we can dangle that incentive.

**Review by arnoldsandoval-okta — APPROVED** [review]

**Pull request merged**
